Chummer/Interfaces/ISustainable.cs

```diff
@@ -0,0 +1,78 @@
+/*  This file is part of Chummer5a.
+ *
+ *  Chummer5a is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Chummer5a is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Chummer5a.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  You can obtain the full source code for Chummer5a at
+ *  https://github.com/chummer5a/chummer5a
+ */
+
+using Chummer.Backend.Skills;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Windows.Forms;
+using System.Xml;
+using NLog;
+
+namespace Chummer
+{
+    public interface ISustainable
+    {
+        /// <summary>
+        /// Force of the spell or complex form
+        /// </summary>
+        int Force { get; set; }
+
+        /// <summary>
+        /// Net Hits of the casting test
+        /// </summary>
+        int NetHits { get; set; }
+
+        /// <summary>
+        /// Is the object sustained by the character?
+        /// </summary>
+        bool SelfSustained { get; set; }
+
+        /// <summary>
+        /// Load an Sustainable Object from an given XML node. (Defaut Path should be sustained/sustainedobject)
+        /// </summary>
+        /// <param name="objNode"></param>
+        void Load(XmlNode objNode);
+
+        /// <summary>
+        /// Print the object's XML to the XmlWriter.
+        /// </summary>
+        /// <param name="objWriter"></param>
+        /// <param name="objCulture"></param>
+        /// <param name="strLanguageToPrint"></param>
+        void Print(XmlTextWriter objWriter, CultureInfo objCulture, string strLanguageToPrint);
+
+        /// <summary>
+        /// Saves an Sustainable object to .xml
+        /// </summary>
+        /// <param name="objWriter"></param>
+        void Save(XmlTextWriter objWriter);
+
+        /// <summary>
+        /// Returns the Display name of an given Spell or Complex Form
+        /// </summary>
+        /// <param name="strLanguage"></param>
+        /// <returns></returns>
+        string DisplayName(string strLanguage);
```

Comment:
Yes they are, this allowed me to keep code cleaner and smaller in other cases, this for example.

```   
           foreach(ISustainable objSustained in SustainedCollection)
           {
               objSustained.Print(objWriter, strLanguageToPrint);
           }
```
Without implementing the `Print()` method in the Interface I would need to distinguish wich `obj` it is beforehand.

The `IHasName` Interface only has the `Name` property. And I needed `DisplayName()` for the same reason I needed the other methods in the interface.

By the way, why don't we use a newer version of C#? The "Default Interface Implementations" sound like an nice feature.

Reply:
> Without implementing the `Print()` method in the Interface I would need to distinguish wich `obj` it is beforehand.

You misunderstand. An interface *should* only contain the methods that it needs to function. A thing that is sustainable does not necessarily have to be printable with a specific CultureInfo and string parameter, for example. What you're probably looking for is a common `IPrintable` interface. Right now, it might seem like needless work to implement `ISustainable` and `IPrintable` as two separate interfaces where all of your `ISustainable` stuff implements `IPrintable` anyway, but it makes sure that those two ways in which a class can function ("Can this be printed?" and "Does this have the properties of a thing that can be sustained?") are separated out and never inadvertently mixed in the future. A similar thing applies to `DisplayName()` stuff.

As for what to do about those right now, I'd probably flag those methods as `\\ TODO: Extract this out into a separate interface` for now, then when this pull request is merged, I'd immediately set up about rectifying that `TODO` in a new one.

> By the way, why don't we use a newer version of C#?

Because newer versions of C# require newer versions of .NET, and newer versions of .NET require newer versions of Windows. .NET Framework 4.7.2 is the most recent version still supported by Windows 7, and we have enough Windows 7 users that switching to a newer version of .NET should only be done if there really is something we can only do in the newer version. E.g., I had us switch from 4.5 to 4.7.2 when I needed 4.7.2's better DPI scaling support to handle a lot of DPI scaling issues.

> The "Default Interface Implementations" sound like an nice feature.

I see it as unnecessary when extension methods exist (and check out the `IHasMatrixAttributes` file to see said extension methods in action when working with an interface). To me, this "feature" really seems like just a more cleaned-up version of setting up an extension method with an interface-casted object as the `this` argument. Sure, the code is cleaner, but this is hardly a reason to switch to a .NET version that is not supported on Windows 7.

Reply:
Not being able to support Win7 anymore is an good argument.

I already reduced the size of that interface by taking out `Print()`, because of a signature mismatch between `Spells` and the other two, after not overriding the complete `Spell.Print()` method anymore. `Load()` also kicked the bucket, because it was unnecessary in the first place.

Will mark the `DisplayName()` and `Save` as `TODO` for now.